vcd/resource_vcd_external_network.go

```diff
@@ -0,0 +1,381 @@
+			DNS1:        ipScopeConfiguration["dns1"].(string),
+			DNS2:        ipScopeConfiguration["dns2"].(string),
+			IPRanges: &types.IPRanges{
+				IPRange: ipRanges,
+			},
+		}
+
+		if ipScopeConfiguration["dns_suffix"] != nil && "" != ipScopeConfiguration["dns_suffix"].(string) {
+			ipScope.DNSSuffix = ipScopeConfiguration["dns_suffix"].(string)
+		}
+
+		ipScopes = append(ipScopes, ipScope)
+	}
+
+	params.Configuration.IPScopes = &types.IPScopes{
+		IPScope: ipScopes,
+	}
+
+	var portGroups []*types.VimObjectRef
+	for _, vsphereNetwork := range d.Get("vsphere_networks").([]interface{}) {
+		portGroup := vsphereNetwork.(map[string]interface{})
+		portGroupConfiguration := &types.VimObjectRef{
+			//MoRef:         portGroup["vsphere_network"].(string),
+			VimObjectType: strings.ToUpper(portGroup["type"].(string)),
+		}
+
+		vCenterHref, err := getVcenterHref(vcdClient.VCDClient, portGroup["vcenter"].(string))
+		if err != nil {
+			return &types.ExternalNetwork{}, fmt.Errorf("unable to find vCenter %s (%#v)", portGroup["vcenter"].(string), err)
+		}
+		portGroupConfiguration.VimServerRef = &types.Reference{
+			HREF: fmt.Sprintf("%s/admin/extension/vimServer/%s", vcdClient.Client.VCDHREF.String(), vCenterHref),
+		}
+
+		var portGroupRecord []*types.PortGroupRecordType
+		if portGroup["type"].(string) == "NETWORK" {
+			portGroupRecord, err = govcd.QueryNetworkPortGroups(vcdClient.VCDClient, portGroup["vsphere_network"].(string))
+		} else {
+			portGroupRecord, err = govcd.QueryDistributedPortGroups(vcdClient.VCDClient, portGroup["vsphere_network"].(string))
+		}
+		if err != nil {
+			return &types.ExternalNetwork{},
+				fmt.Errorf("unable to find port group %s (%#v)", portGroup["vcenter"].(string), err)
+		}
+
+		if len(portGroupRecord) == 1 {
+			portGroupConfiguration.MoRef = portGroupRecord[0].MoRef
+		} else {
+			return &types.ExternalNetwork{},
+				fmt.Errorf("find less or more than one port groups - %d", len(portGroupRecord))
+		}
+
+		portGroups = append(portGroups, portGroupConfiguration)
+	}
+	params.VimPortGroupRefs = &types.VimObjectRefs{
+		VimObjectRef: portGroups,
+	}
+
+	if parentNetworkName, ok := d.GetOk("parent_network"); ok {
+		name := parentNetworkName.(string)
+		parentNetwork, err := govcd.GetExternalNetworkByName(vcdClient.VCDClient, name)
+		if err != nil {
+			return &types.ExternalNetwork{}, fmt.Errorf("unable to find parent network %s (%s)", name, err)
+		}
+		params.Configuration.ParentNetwork = &types.Reference{
+			HREF: parentNetwork.HREF,
+			Type: parentNetwork.Type,
+			Name: parentNetwork.Name,
+		}
+	}
+
+	if description, ok := d.GetOk("description"); ok {
+		params.Description = description.(string)
+	}
+
+	return params, nil
+}
+
+func getVcenterHref(vcdClient *govcd.VCDClient, name string) (string, error) {
+	virtualCenters, err := govcd.QueryVirtualCenters(vcdClient, fmt.Sprintf("(name==%s)", name))
+	if err != nil {
+		return "", err
+	}
+	if len(virtualCenters) == 0 {
```

Comment:
Not sure if it can be the case, but should we protect from finding multiple vCenters with the same name?

Something like:
```suggestion
	if len(virtualCenters) == 0 || len(virtualCenters) > 1 {
```

Reply:
To be sure we can add.